### ISSUE TITLE
Fresh installs of Lokole 0.1.40+ fail during pip, on Raspbian

### ISSUE THREAD

**Issue by floydianslips** [iiab/iiab#1638]
### Specifications like the version of the IIAB, operating system version, or hardware details.

Raspberry Pi 3 B+
Raspbian Lite (2019-04-08)
latest pre-release of [IIAB 7.0](https://github.com/iiab/iiab/wiki/IIAB-7.0-Release-Notes) (master branch, as auto-installed by official http://d.iiab.io)

![Lokole-error](https://user-images.githubusercontent.com/8054387/57347002-9fb93100-7105-11e9-9493-e7de45fd5c39.png)

**Comment by floydianslips** [comment]
```
$ sudo pip install opwen_email_client
Collecting opwen_email_client
  Using cached https://files.pythonhosted.org/packages/1b/00/0e4c46c1d419d7e59ef8ef5fe91999cecd16b4b46ed2e950d8c01df2bb70/opwen_email_client-0.1.41.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-REAkkZ/opwen-email-client/setup.py", line 4, in <module>
        with open('requirements.txt', encoding='utf-8') as fobj:
    TypeError: 'encoding' is an invalid keyword argument for this function
    Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-REAkkZ/opwen-email-client/

```

**Comment by holta** [comment]
Does the same error occur if you change [/opt/iiab/iiab/roles/lokole/defaults/main.yml#L8](https://github.com/iiab/iiab/blob/master/roles/lokole/defaults/main.yml#L8) from 0.1.41 to 0.1.38 as per IIAB 6.7 &mdash; and then run `cd /opt/iiab/iab` followed by `./runrole lokole` ?

Ref: #1630 "Validate Lokole 0.1.41"

**Comment by floydianslips** [comment]
Changing the version to 0.1.38 [on line 8 of /opt/iiab/iiab/lokole/defaults/main.yml] allowed `./runrole lokole` to complete without an issue.

**Comment by floydianslips** [comment]
Installed 0.1.38 and then in order _[upgraded 3 times, using `./runrole lokole` each time, to]_ .39, .40 and .41 without issue.

**Comment by holta** [comment]
@c-w @aidan-fitz

> Installed 0.1.38 and then in order .39, .40 and .41 without issue.

Many thanks @floydianslips: this confirms the regression in Lokole occurs somewhere after 0.1.38 &mdash; regardless whether installing on Raspbian Lite or Raspbian Desktop.

**Comment by floydianslips** [comment]
Initial install of 0.1.38 and 0.1.39 seem to install without issue. 0.1.40 and 0.1.41 have the same error as above.

**Comment by holta** [comment]
> Initial install of 0.1.38 and 0.1.39 seem to install without issue. 0.1.40 and 0.1.41 have the same error as above.

@c-w and @aidan-fitz see above &mdash; the regression was not part of 0.1.39 &mdash; but definitely appears to affect Lokole 0.1.40+

We're hopeful you can take the investigation from here, so the most recent versions of Lokole might once again be installable on Raspberry Pi before June, when IIAB 7.0's final release is scheduled?

PS I just reverted IIAB's master branch [from 0.1.41 to 0.1.39](https://github.com/iiab/iiab/commit/bf9e96c8cf08cfe51e6013e3b222804af5e115ff) so that [BIG-sized] IIAB installs once again now work!

**Comment by holta** [comment]
I had repeated problems pip installing Lokole 0.1.39 on Raspbian Desktop on RPi 3 B+ earlier today.

Reverting to a fresh install of 0.1.38 seems to have worked.  I don't get it, as I thought the issue was isolated to fresh installs of 0.1.49+

Thanks all for explanations, experiences, and possible workarounds...

**Comment by c-w** [comment]
I believe this is a bug in `setup.py` that was introduced in https://github.com/ascoderu/opwen-webapp/commit/d0159dde193504a22a96b6ba6d4c5fb87c03acb2 and fixed in https://github.com/ascoderu/opwen-webapp/commit/c9aebc03c159530153189925462bbd01c201ff97. A fix was published in [0.1.42](https://pypi.org/project/opwen-email-client/0.1.42/). Apologies for the inconvenience and thanks for your patience for the fix!

**Comment by holta** [comment]
> I believe this is a bug in `setup.py` that was introduced in [ascoderu/opwen-webapp@d0159dd](https://github.com/ascoderu/opwen-webapp/commit/d0159dde193504a22a96b6ba6d4c5fb87c03acb2) and fixed in [ascoderu/opwen-webapp@c9aebc0](https://github.com/ascoderu/opwen-webapp/commit/c9aebc03c159530153189925462bbd01c201ff97). A fix was published in [0.1.42](https://pypi.org/project/opwen-email-client/0.1.42/). Apologies for the inconvenience and thanks for your patience for the fix!

Thanks @c-w !

Any reason that IIAB 7.0 (final release expected in a few weeks, in June) should not consider Lokole 0.4.0 from https://github.com/ascoderu/opwen-webapp/releases ?

(On Raspbian / ARM, Ubuntu 18.04, Debian 9 and ideally Debian 10 too :)

**Comment by c-w** [comment]
@holta Latest version `0.4.0` should be fine. There were some breaking changes to the sync setup and cronjob setup, but none of those features are exercised in the IIAB mode.

**Comment by holta** [comment]
Appears fixed by PR #1666 &mdash; thanks everybody for [IIAB 7.0](https://github.com/iiab/iiab/wiki/IIAB-7.0-Release-Notes) pre-release testing as we approach release in coming weeks!
